zookeeper-server/src/main/java/org/apache/zookeeper/cli/DelQuotaCommand.java

```diff
@@ -43,11 +42,14 @@
     private CommandLine cl;
 
     public DelQuotaCommand() {
-        super("delquota", "[-n|-b] path");
+        super("delquota", "[-n|-b|-N|-B] path");
 
         OptionGroup og1 = new OptionGroup();
-        og1.addOption(new Option("b", false, "bytes quota"));
-        og1.addOption(new Option("n", false, "num quota"));
+        og1.addOption(new Option("n", false, "num soft quota"));
+        og1.addOption(new Option("b", false, "bytes soft quota"));
+        og1.addOption(new Option("N", false, "num hard quota"));
+        og1.addOption(new Option("B", false, "bytes hard quota"));
```

Comment:
Not a bug you introduced—it's also in `master`—but it turns out that that `OptionGroup` business does not quite work as one would expect:

```
] create /foo
Created /foo
] setquota -b 42 /foo
```

Okay.

```
] create /bar
Created /bar
] setquota -b 42 /bar
```

Still okay, because we're using the `-b` flag again.   But this:

```
] create /baz
Created /baz
] setquota -B 42 /baz
org.apache.commons.cli.AlreadySelectedException: The option 'B' was specified but an option from this group has already been selected: 'b'
```

… now fails, because the "global" `OptionGroup` object has been modified to mark the `-b` as in use!  I'd suggest getting rid of the `option` field, and recreating the `Option` "tree" for each parse.

A workaround is to quit and restart the "cli" program.

Reply:
@ztzg 
- I know this. It may be not a bug, just a feature:). A little inconvenient when changing the option.
- Change `OptionGroup` to `Option` will survive from it? e.g: `SetAclCommand`

Reply:
I should have been a bit clearer, because unless I missed something, once one has used the `-b` flag of `setquota`, for example, one can *never, ever again* use another flag with `setquota`, *even on another node*, unless one exits and restarts `zkCli.sh`. That seems a bit extreme as a "feature" :)

(But again: you did not introduce that bug; it was there before.  I can fix it in another ticket/PR.)

Reply:
- Yes it's doable. Please create another ticket for it.
- using `OptionGroup `will have that inconvenient, which making options exclusive in one client session. 
- For the fix, no need to restart? just replace `OptionGroup `to `Option`, then check the args length in the code?
- You can also create a ticket for this issue you found:
`if (f (parentName.startsWith(prh(procZookeeper) &&Quotas.limitNode.equals(chi(childName)) `

Reply:
Okay—I'll take care of those :)